src/service_objects/game_piece_action_resolvers/ranged_attack_resolver.ts

```diff
@@ -82,8 +90,9 @@ export default async function resolveRangedAttackAction(
   }
 
   // Update action record as resolved
-  actionData.resolved = true;
-  actionData.resolvedAttacks = resolvedAttacks;
-  action.actionData = actionData;
+  let newActionData = JSON.parse(JSON.stringify(actionData));
+  newActionData.resolved = true;
+  newActionData.resolvedAttacks = resolvedAttacks;
+  action.actionData = newActionData;
```

Comment:
Needed to clone the `actionData` JSON object because it was not saving to the record correctly.